src/app/pages/ValidatorDetailsPage/BalanceDistributionCard.tsx

```diff
@@ -1,13 +1,60 @@
 import { FC } from 'react'
 import { useTranslation } from 'react-i18next'
-import { SnapshotTextCard } from '../../components/Snapshots/SnapshotCard'
+import { TFunction } from 'i18next'
+import { Validator } from '../../../oasis-nexus/api'
+import { getPreciseNumberFormat } from '../../../locales/getPreciseNumberFormat'
+import { SnapshotCard } from '../../components/Snapshots/SnapshotCard'
+import { PieChart } from '../../components/charts/PieChart'
 
-export const BalanceDistributionCard: FC = () => {
+type BalanceDistributionCardProps = {
+  validator?: Validator
+}
+
+function chartData(t: TFunction, validator: Validator | undefined) {
+  if (
+    validator?.escrow?.active_balance === undefined ||
+    validator?.escrow?.self_delegation_balance === undefined
+  ) {
+    return []
```

Comment:
A little odd that you're checking if active_balance exists, but then using active_shares

Reply:
forgot to update while switching to shares after discussion with Don and Andy